lib/MusicBrainz/Server/Controller/Release.pm

```diff
@@ -386,6 +388,61 @@ sub _merge_parameters {
     }
 }
 
+sub _merge_on_creation {
+    my ($self, $c, $edit, $form, $new, $old_ids, $entity_id) = @_;
+    if ($form->field('merge_rgs')->value) {
+        my $edit_id = $edit->id;
+
+        my $new_rg_id = $new->release_group->id;
+        # We want to make sure we're merging only different RGs!
+        my @old_entities = map +{ id => $_->id, name => $_->name },
+            grep { $_->id != $new_rg_id }
+            map { %$entity_id{$_}->release_group }
```

Comment:
```suggestion
            map { $entity_id->{$_}->release_group }
```
(IMO this variable should be renamed to `entities_by_id` or something similar, too.)